curriculum/challenges/english/25-front-end-development/quiz-javascript-functional-programming/66edd4f31ff19bf5573bf64b.md

```diff
@@ -17,439 +17,472 @@ Answer all of the questions below correctly to pass the quiz.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In the context of pure functions, what does "referential transparency" mean?
 
 #### --distractors--
 
-Placeholder distractor 1
+The function's code is visible to other parts of the program
 
 ---
 
-Placeholder distractor 2
+The function references external variables transparently
 
 ---
 
-Placeholder distractor 3
+The function's name clearly indicates its purpose
 
 #### --answer--
 
-Placeholder answer
+The function can be replaced with its corresponding value without changing the program's behavior
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is NOT a benefit of using pure functions?
 
 #### --distractors--
 
-Placeholder distractor 1
+Easier to test
 
 ---
 
-Placeholder distractor 2
+More predictable behavior
 
 ---
 
-Placeholder distractor 3
+Easier to reason about
 
 #### --answer--
 
-Placeholder answer
+Improved performance in all cases
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the output of this pure function?
+```javascript
+const add = (a, b) => a + b;
+console.log(add(2, 5));
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+3
 
 ---
 
-Placeholder distractor 2
+4
 
 ---
 
-Placeholder distractor 3
+undefined
 
 #### --answer--
 
-Placeholder answer
+7
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which statement about currying is true?
 
 #### --distractors--
 
-Placeholder distractor 1
+Currying always improves performance
 
 ---
 
-Placeholder distractor 2
+Curried functions are always pure functions
 
 ---
 
-Placeholder distractor 3
+Currying is only possible in functional programming languages
 
 #### --answer--
 
-Placeholder answer
+Currying transforms a function with multiple arguments into a sequence of functions
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is a potential side effect of the following function?
+```javascript
+const greet = (name) => {
+  console.log(`Hello, ${name}!`);
+};
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+It modifies a global variable
 
 ---
 
-Placeholder distractor 2
+It changes the input parameter
 
 ---
 
-Placeholder distractor 3
+It returns a value
 
 #### --answer--
 
-Placeholder answer
+It interacts with the console (I/O operation)
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In functional programming, what is function composition?
 
 #### --distractors--
 
-Placeholder distractor 1
+Writing functions with multiple arguments
 
 ---
 
-Placeholder distractor 2
+Declaring functions inside other functions
 
 ---
 
-Placeholder distractor 3
+Using arrow function syntax
 
 #### --answer--
 
-Placeholder answer
+Combining two or more functions to create a new function
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is an example of a pure function?
+```javascript
+const double = (x) => x * 2;
+```
```

Comment:
````suggestion
Which of the following is an example of a pure function?

```js
const double = (x) => x * 2;
```
````
should this code be in one of the distractors/answers or is it part of the question?